src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2821,6 +2825,50 @@ SqlTaskList(Job *job)
 }
 
 
+/*
+ * RelabelTypeMutator walks over the tree and converts RelabelType's
+ * into CollationExpr's. With that, we will be able to pushdown COLLATE's.
+ */
+static Node *
+RelabelTypeMutator(Node *originalNode)
+{
+	Node *newNode = NULL;
+	if (originalNode == NULL || !IsA((Query *) originalNode, Query))
+	{
+		return NULL;
+	}
+
+	if (IsA(originalNode, RelabelType))
+	{
+		RelabelType *originalRelabelType = (RelabelType *) originalNode;
+		if (OidIsValid(originalRelabelType->resultcollid))
+		{
+			CollateExpr *newCollateExpr = makeNode(CollateExpr);
+			newCollateExpr->arg = originalRelabelType->arg;
+			newCollateExpr->collOid = originalRelabelType->resultcollid;
+			newCollateExpr->location = originalRelabelType->location;
+			newCollateExpr->xpr = originalRelabelType->xpr;
+			return (Node *) expression_tree_mutator((Query *) originalRelabelType,
```

Comment:
I think we should call `expression_tree_mutator` with `newCollateExpr` instead of `originalRelabelType`, not %100 sure